src/chess/board.cc

```diff
@@ -1212,4 +1170,63 @@ Move ChessBoard::ParseMove(std::string_view move_str) const {
   return Move::White(from, to);
 }
 
+std::string BoardToFen(const ChessBoard& board) {
```

Comment:
We already have `GetFen()` in position.h/cc, we don't need both. From a quick look `GetFen()` seems cleaner, but I like the `BoardToFen()` interface better.

Reply:
Obviously `BoardToFen()` can't output move count and rule50 plies, so we probably need to make `GetFen()` call `BoardToFen()`.

Reply:
I remembered we have this, but couldn't find!

Did a few changes:
- Made GetPieceAt take square instead of ints.
- Moved the code (with some further cleanup) to BoardToFen()
- Renamed GetFen() to PositionToFen(), and made it call BoardToFen() plus append two more fields.
- Made DebugString() of both Position and Board to call corresponding *ToFen() func, also removed the hash output.